app/cdap/components/DirectiveInput/Components/InputPanel/index.tsx

```diff
@@ -0,0 +1,228 @@
+import { IDirectiveUsage, IObject } from 'components/DirectiveInput/types';
+import ee from 'event-emitter';
+import Fuse from 'fuse.js';
+import reverse from 'lodash/reverse';
+import Mousetrap from 'mousetrap';
+import React, { useEffect, useState } from 'react';
+import globalEvents from 'services/global-events';
+import NamespaceStore from 'services/NamespaceStore';
+import styled from 'styled-components';
+import uuidV4 from 'uuid/v4';
+import { grey } from '@material-ui/core/colors';
+import { IHeaderNamesList } from 'components/GridTable/types';
+
+interface IInputPanelProps {
+  setDirectivesList: React.Dispatch<React.SetStateAction<[]>>;
+  directiveSelected: boolean;
+  columnNamesList: IHeaderNamesList[];
+  onSearchItemClicked: (value: string) => void;
+  getDirectiveSyntax: (results: IDirectiveUsage[], value: boolean) => void;
+  onColumnSelected: (value: boolean) => void;
+  inputBoxValue: string;
+}
+
+const SimpleWrapper = styled(Box)`
+  display: block;
+`;
+
+const SmallLabel = styled(Typography)`
+  font-style: normal;
+  font-weight: 400;
+  font-size: 14px;
+  letter-spacing: 0.15;
+  color: ${grey[700]};
+`;
+
+const LargeLabel = styled(SmallLabel)`
+  font-weight: 600;
+  font-size: 16px;
+`;
+
+const ResultRow = styled(Box)`
+  padding: 10px;
+  border-bottom: 1px solid ${grey[300]};
+  background-color: ${(props) => (props.isActive ? '#EFF0F2' : '#FFFFFF')};
+  &:hover {
+    background: #eff0f2;
+    cursor: pointer;
+  }
+`;
+
+export default function({
+  setDirectivesList,
+  directiveSelected,
+  columnNamesList,
+  onSearchItemClicked,
+  getDirectiveSyntax,
+  onColumnSelected,
+  inputBoxValue,
+}: IInputPanelProps) {
+  const [searchResults, setSearchResults] = useState<IDirectiveUsage[]>([]);
+  const [inputText, setInputText] = useState<string>('');
+  const [selectedIndex, setSelectedIndex] = useState<number | null>(null);
+  const eventEmitter = ee(ee);
+  const [fuse, setFuse] = useState(new Fuse([], { ...defaultFuseOptions }));
+
+  const getUsage = () => {
+    if (!directiveSelected) {
+      MyDataPrepApi.getUsage({ context: NamespaceStore.getState().selectedNamespace }).subscribe(
+        (res) => {
+          setDirectivesList(res.values);
+          setFuse(new Fuse(res.values, { ...defaultFuseOptions }));
+        }
+      );
+    } else {
+      setFuse(new Fuse(columnNamesList, { ...defaultFuseOptions, keys: ['label'] }));
+    }
+  };
+
+  useEffect(() => {
+    getUsage();
+  }, [directiveSelected]);
+
+  useEffect(() => {
+    eventEmitter.on(globalEvents.DIRECTIVEUPLOAD, getUsage);
+    const directiveInput = document.getElementById('directive-input-search');
+    const mousetrap = new Mousetrap(directiveInput);
+    mousetrap.bind('up', handleUpArrow);
+    mousetrap.bind('down', handleDownArrow);
+    mousetrap.bind('enter', handleEnterKey);
+    mousetrap.bind('tab', handleTabKey);
+
+    // unbind a single keyboard event or an array of keyboard events.
```

Comment:
which ones are single events
which ones are array of events

plz break both and place in proper line

Reply:
Done